extension/src/experiments/webview/messages.ts

```diff
@@ -248,6 +245,29 @@ export class WebviewMessages {
     }
   }
 
+  private async createAndSelectBranchFromExperiment(expId: string) {
+    const branchName = await commands.executeCommand(
+      RegisteredCliCommands.EXPERIMENT_VIEW_BRANCH,
+      { dvcRoot: this.dvcRoot, id: expId }
+    )
+
+    if (!branchName) {
+      return
+    }
+
+    await this.update()
```

Comment:
We have to run `update` twice (once to make sure the branch is in `availableBranchesToSelect` and another time to update the table). 

We could run only branch updates on the first run (aka run only `data.updateBranches()` which is currently `private`) though this would mean more code to maintain.

Reply:
I don't think you need this. You can get the current list of selected branches with `this.experiments.getSelectedBranches()`
and push the new branch in and use `setSelectedBranches` before calling `update`. If it does not exist then it will be filtered out, if it is there it should be fine.

Reply:
```
  public setSelectedBranches(branches: string[]) {
    this.selectedBranches = branches.filter(
      branch => branch !== this.currentBranch
    )
    this.persistBranchesToShow()
  }
```
☝🏻 there is nothing to filter the new branch here

```
  public setBranches(
    branches: string[],
    branchesToSelect: string[],
    currentBranch: string
  ) {
    this.availableBranchesToShow = branches
    this.availableBranchesToSelect = branchesToSelect
    this.currentBranch = currentBranch

    this.selectedBranches = this.selectedBranches.filter(
      branch => branches.includes(branch) && branch !== this.currentBranch
    )
    this.persistBranchesToShow()
  }
```
☝🏻 but there is here

```
  private async updateBranches() {
    const allBranches = await this.internalCommands.executeCommand<string[]>(
      AvailableCommands.GIT_GET_BRANCHES,
      this.dvcRoot
    )

    const { currentBranch, branches, branchesToSelect } =
      collectBranches(allBranches)

    this.experiments.setBranches(branches, branchesToSelect, currentBranch)
  }
```
so inserting the branch before calling update (which calls the above) should be fine.

.... please nail that down with a test

Reply:
> You can get the current list of selected branches with this.experiments.getSelectedBranches()
and push the new branch in and use setSelectedBranches before calling update. If it does not exist then it will be filtered out, if it is there it should be fine.

We could do that! However we will need to `.sort()` the branch list to ensure that the branches are in the right order.

Reply:
Adjusted the code to just add the new branch with `setBranches` and `getSelectedBranches`!